apiserver/facades/client/bundle/bundle.go

```diff
@@ -380,27 +380,42 @@ func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
 		return fail(err)
 	}
 
+	// First create a bundle output from the bundle data.
 	var buf bytes.Buffer
 	enc := yaml.NewEncoder(&buf)
+	if err != nil {
+		return fail(err)
+	}
 	if err = enc.Encode(bundleOutputFromBundleData(base)); err != nil {
 		return fail(err)
-	} else if err = enc.Encode(overlay); err != nil {
+	}
+
+	// Secondly create an output from the overlay. We do it this way, so we can
+	// insert the correct comments for users.
+	output := buf.String()
+	buf.Reset()
+	if err = enc.Encode(overlay); err != nil {
 		return fail(err)
 	} else if err = enc.Close(); err != nil {
 		return fail(err)
 	}
+	overlayOutput := buf.String()
 
-	// If the overlay part is empty, strip it off; otherwise, inject a
+	// If the overlay part is empty, ignore it; otherwise, inject a
 	// comment to let users know that the second document can be extracted
 	// out and used as a standalone overlay.
-	yamlOut := buf.String()
-	if strings.HasSuffix(yamlOut, "--- {}\n") {
-		yamlOut = yamlOut[:strings.Index(yamlOut, "---")]
-	} else {
-		yamlOut = strings.Replace(yamlOut, "---", "--- # overlay.yaml", 1)
+	if !strings.HasPrefix(overlayOutput, "--- {}\n") {
+		// strip off the first three dashes and merge the base bundle and the
+		// overlay.
+		if strings.HasPrefix(overlayOutput, "---") {
+			overlayOutput = strings.Replace(overlayOutput, "---", "--- # overlay.yaml", 1)
+			output = output + overlayOutput
+		} else {
+			return fail(errors.Errorf("unexpected overlay"))
```

Comment:
This would indicate a bug with goyaml. Should we make this a bit more explicit (e.g. `BUG: expected yaml encoder to delineate multiple documents with "---" separator`) so that if this ever happens, the person assigned to the bug does not go on a wild goose chase trying to debug the bundledata -> {base, overlay} code in charm.v6?